src/librustc_resolve/diagnostics.rs

```diff
@@ -651,7 +661,7 @@ impl<'a> Resolver<'a> {
         let suggestion = self.early_lookup_typo_candidate(
             ScopeSet::Macro(macro_kind), parent_scope, ident, is_expected
         );
-        add_typo_suggestion(err, suggestion, ident.span);
+        add_typo_suggestion(err, suggestion, ident.span, None);
```

Comment:
The definition span should be obtainable here in the same way as in `lookup_typo_candidate` below.
So the `def_span` logic can be moved entirely into `add_typo_suggestion`.